frontend/src/utils/account-send.ts

```diff
@@ -0,0 +1,144 @@
+import { isAddress, keccak256, toUtf8Bytes, BigNumber, hexZeroPad } from 'src/utils/ethers';
+import localforage from 'localforage';
+import { toAddress, lookupAddress } from 'src/utils/address';
+import { LOCALFORAGE_ACOUNT_SEND_KEY } from 'src/utils/constants';
+import { Web3Provider } from 'src/utils/ethers';
+
+type AccountData = {
+  address: string;
+  advancedMode: boolean;
+  checkbox: boolean;
+};
+
+type KeyData = {
+  count: number;
+  viewingKey: string;
+};
+
+type EncryptedData = {
+  encryptedAddress: string;
+};
+
+type AccountSendData = {
+  amount: string;
+  tokenAddress: string;
+  dateSent: Date;
+  hash: string;
+  userAddress: string;
+  recipientAddress: string;
+  checkbox?: boolean;
+  advancedMode?: boolean;
+};
+
+type DecryptedSendData = {
+  recipientId: string;
+};
+
+const accountData = ({ address, advancedMode, checkbox }: AccountData) => {
+  if (!isAddress(address)) {
+    throw new Error('Invalid address');
+  }
+  const advancedModeBit = advancedMode ? '1' : '0';
+  const checkboxBit = checkbox ? '1' : '0';
+  const addressHex = BigNumber.from(address).toHexString() + advancedModeBit + checkboxBit;
+
+  return BigNumber.from(hexZeroPad(addressHex, 32));
+};
+
+const encryptAccountData = ({ address, advancedMode, checkbox, count, viewingKey }: AccountData & KeyData) => {
+  const key = keccak256(toUtf8Bytes(`${viewingKey}${count}`));
+  const data = accountData({ address, advancedMode, checkbox });
+  const encryptedData = data.xor(key);
+  return hexZeroPad(encryptedData.toHexString(), 32);
+};
+
+export const decryptData = ({ viewingKey, count, encryptedAddress }: KeyData & EncryptedData) => {
```

Comment:
We should definitely have some tests for these methods in the `frontend/test` folder to make sure we don't break things in the future

Reply:
Is it okay if we leave `fetchAccountSends` and `storeSend`? My worry with testing them heavily is that the tests would be brittle as they would be heavily dependent on the underlying encryption and decryption methods, and these functions are mainly meant to format data.

Reply:
Added tests for encrypt and decrypt https://github.com/ScopeLift/umbra-protocol/pull/494/commits/67b1b5d4b80da11719360901bbfa04b1657517ca

Reply:
Awesome. Before resolving this I want to take a closer look and make sure they cover all the edge cases